src/components/Reports/ReportsDailyVisits/ReportsDailyVisits.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage } from 'react-intl';
+import Loading from '../../Loading/Loading';
+import C3Chart from '../../shared/C3Chart/C3Chart';
+
+let chartConfig = {
+  data: {
+    json: null,
+    type: 'spline',
+    keys: {
+      x: 'from',
+      value: ['quantity'],
+    },
+  },
+  legend: {
+    show: false,
+  },
+  axis: {
+    x: {
+      type: 'timeseries',
+      tick: {
+        format: '%Y-%m-%d',
+      },
+    },
+  },
+};
+
+const ReportsDailyVisits = ({ domainName, dateFrom, dependencies: { datahubClient } }) => {
+  const [state, setState] = useState({ loading: true });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const dailyVisitsData = await datahubClient.getDailyVisitsByPeriod({
+        domainName: domainName,
+        dateFrom: dateFrom,
+      });
+      if (!dailyVisitsData.success || !dailyVisitsData.value) {
+        setState({ loading: false });
+      } else {
+        const data = dailyVisitsData.value.map((item) => {
+          let date = new Date(item.from);
+          date = date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + date.getDate();
+          return {
+            quantity: item.quantity,
+            from: date,
+          };
+        });
+        chartConfig.data.json = data;
+        setState({
+          loading: false,
+          chartConfig: chartConfig,
+        });
+      }
+    };
+
+    fetchData();
+  }, [datahubClient, dateFrom, domainName]);
+
+  return (
+    <div className="wrapper-reports-box">
+      <div className="reports-box">
+        <small className="title-reports-box">
+          <FormattedMessage id="reports_daily_visits.title" />
+        </small>
+        {state.loading ? (
+          <Loading />
+        ) : (
+          <>
+            <div id="dailyVisitsChart"></div>
+            {<C3Chart config={state.chartConfig} element="dailyVisitsChart" />}
```

Comment:
What is the meaning of these curly braces?

```suggestion
            <C3Chart config={state.chartConfig} element="dailyVisitsChart" />
```

Reply:
Is `dailyVisitsChart` the Id of an element? it should not be explicit.

Reply:
is the id of element right

Reply:
Mmmm, creo que esta no es una buena API para el componente, en mi opinión debería ser así:

```diff
- <>
-   <div id="dailyVisitsChart"></div>
-   <C3Chart config={state.chartConfig} element="dailyVisitsChart" />
+   <C3Chart config={state.chartConfig} />
- </>
```

Reply:
en el config pasar el id del elemento?

Reply:
No, creo que deberíamos evitar totalmente hacer otro elemento y pasar Ids.

Reply:
listo